amm/drink-tests/src/stable_swap_tests/tests_add_remove_lp.rs

```diff
@@ -0,0 +1,400 @@
+use drink::{self, session::Session};
+
+use super::*;
+
+// ref https://github.com/ref-finance/ref-contracts/blob/d241d7aeaa6250937b160d56e5c4b5b48d9d97f7/ref-exchange/tests/test_stable_pool.rs#L123
+#[drink::test]
+fn test_01(mut session: Session) {
+    seed_account(&mut session, CHARLIE);
+    seed_account(&mut session, DAVE);
+    seed_account(&mut session, EVA);
+    let initial_reserves = vec![100000 * ONE_DAI, 100000 * ONE_USDT, 100000 * ONE_USDC];
+    let (stable_swap, tokens) = setup_stable_swap_with_tokens(
+        &mut session,
+        vec![18, 6, 6],
+        initial_reserves
+            .iter()
+            .map(|amount| amount * 100_000_000_000)
+            .collect(),
+        10_000,
+        25,
+        2000,
+        BOB,
+    );
+
+    _ = stable_swap::add_liquidity(
+        &mut session,
+        stable_swap,
+        BOB,
+        1,
+        initial_reserves.clone(),
+        bob(),
+    )
+    .expect("Should successfully add liquidity");
+
+    let (last_share_price, last_total_shares) =
+        share_price_and_total_shares(&mut session, stable_swap, None);
+
+    transfer_and_increase_allowance(
+        &mut session,
+        stable_swap,
+        tokens.clone(),
+        CHARLIE,
+        vec![500 * ONE_DAI, 500 * ONE_USDT, 500 * ONE_USDC],
+        BOB,
+    );
+
+    // add more liquidity with balanced tokens (charlie)
+    _ = stable_swap::add_liquidity(
+        &mut session,
+        stable_swap,
+        CHARLIE,
+        1,
+        vec![500 * ONE_DAI, 500 * ONE_USDT, 500 * ONE_USDC],
+        charlie(),
+    )
+    .expect("Should successfully add liquidity");
+
+    assert_eq!(
+        share_price_and_total_shares(&mut session, stable_swap, None),
+        (last_share_price, last_total_shares + 1500 * ONE_LPT)
```

Comment:
Do we need the `last_share_price`? What we're really checking is that reserves increased by 500 each and that there's 3*1500 more LP tokens. What I want to say is that we're really testing that reserves/LP tokens increase as expected. Maybe that's worth extracting to a standalone test? Maybe I'm overreacting but I like small and simple tests that test one thing only (if possible).